book/views.py

```diff
@@ -49,3 +64,33 @@ def get(self, request, book_id):
         except Book.DoesNotExist:
             return JsonResponse({'message':'NOT_EXIST_BOOK'}, status=400)
 
+class SearchBookView(View):
+    def get(self, request):
+        try:
+            total   = request.GET.get('total', None)
+            author  = request.GET.get('author', None)
+            title   = request.GET.get('title', None)
+            company = request.GET.get('company', None)
+
+            json_data = None
+            books = None
+            if author:
+                books = Book.objects.filter(author__icontains = author)
+            elif title:
+                books = Book.objects.filter(title__icontains = title)
+            elif company:
+                books = Book.objects.filter(company__icontains = company)
+            elif total:
+                books = Book.objects.filter(Q(author__icontains=total)|
+                                            Q(title__icontains=total)|
+                                            Q(company__icontains=total))
+            else: 
+                return JsonResponse({"message":"INVALID_REQUEST"}, status=400)
+                
+            json_data = list(books.values('id', 'author', 'title', 'image_url', 'company'))
+            
+            return JsonResponse({"message":"SUCCESS", "books":json_data}, status=200)
+
```

Comment:
```suggestion
                    return JsonResponse({"message":"SUCCESS", "books":json_data}, status=200)

            else: 
                return JsonResponse({"message":"INVALID_REQUEST"}, status=400)
```
코드의 흐름상 200 ok가 더 위로 올라가서 로직과 더 가까이 붙어있는 것이 좋을 것 같아요 👍